pkg/cloudprovider/aws/fleet/launchtemplate.go

```diff
@@ -142,27 +143,33 @@ func (p *LaunchTemplateProvider) getSecurityGroupIds(ctx context.Context, cluste
 	return securityGroupIds, nil
 }
 
-func (p *LaunchTemplateProvider) getAMIID(context context.Context) (*string, error) {
+func (p *LaunchTemplateProvider) getAMIID(ctx context.Context) (*string, error) {
 	version, err := p.kubeServerVersion()
 	if err != nil {
 		return nil, fmt.Errorf("kube server version, %w", err)
 	}
-	paramOutput, err := p.ssm.GetParameter(&ssm.GetParameterInput{
-		Name: aws.String(fmt.Sprintf("/aws/service/eks/optimized-ami/%s/amazon-linux-2/recommended", version)),
+	paramOutput, err := p.ssm.GetParameterWithContext(ctx, &ssm.GetParameterInput{
+		Name: aws.String(fmt.Sprintf("/aws/service/bottlerocket/aws-k8s-%s/x86_64/latest/image_id", version)),
 	})
 	if err != nil {
-		return nil, fmt.Errorf("ssm get parameter, %w", err)
+		return nil, fmt.Errorf("getting ssm parameter, %w", err)
 	}
-	output := struct {
-		ImageID string `json:"image_id"`
-	}{}
-	if err := json.Unmarshal([]byte(*paramOutput.Parameter.Value), &output); err != nil {
-		return nil, fmt.Errorf("unmarshal parameter output, %w", err)
+	return paramOutput.Parameter.Value, nil
+}
+
+func (p *LaunchTemplateProvider) getUserData(cluster *v1alpha1.ClusterSpec) (*string, error) {
+	t := template.Must(template.New("bottlerocketUserData").Parse(bottleRocketUserData))
```

Comment:
nit: I dislike needing to name template (bad api imo), but i'd simplify to something smaller to cut down on code bloat. Maybe "userdata"? Minimally, with your approach bottlerocketUserData and bottleRocketUserData should use the same casing.